tests/manage/external/test_postgres.py

```diff
@@ -60,3 +66,15 @@ def test_create_drop_db(self):
         self.assertTrue(postgres.db_exists(self.dbname))
         postgres.drop_db(self.dbname)
         self.assertFalse(postgres.db_exists(self.dbname))
+
+    def test_create_dbuser_db_safe(self):
+        """Test creating a database and user in a safe way"""
+        postgres = self._setup_postgres()
+        user1, db1 = postgres.create_dbuser_db_safe(self.dbname, self.dbuser, self.dbpass)
+        assert user1 == self.dbuser
+        assert db1 == self.dbname
+
+        # a second try should reuse the database user but create a new database
+        user2, db2 = postgres.create_dbuser_db_safe(self.dbname, self.dbuser, self.dbpass)
+        assert user2 == self.dbuser
+        assert db2 == f'{self.dbname}_1'
```

Comment:
Could we add an additional case here where you pass an invalid password to see it creates a new user automatically? Or do we hit the same problem potentially where certain postgres servers trust local connections? Is this the case for our CI setup? If not, I would still add the test (as well as the one you commented out) just so we have more coverage. If it then fails when people run the tests locally and they have their postgres configured that way, that is ok I think.

Reply:
Hm... I've thought about it but is it really worth adding a test here that may fail depending on the details of the test setup?

The only thing this test would add is to check that psycopg2 throws the same exception when the password is wrong as when the user name is wrong (wrong user name is already tested).
I've checked that manually, and it does.

Reply:
What I mean though is that lines 221-224 of `create_dbuser_db_safe` are never hit in this test. In the first test, the user doesn't exist, and so the first conditional is hit and it is created. In the second test, the user exists and it can authenticate, so it goes straight to the database creation, skipping the block with `self.find_new_user`. So is that actually getting covered now by the tests? And my point here is that it is better to have that test covered in our CI and have it potentially fail if some developer runs the test locally and has a weird Postgres configuration. Anyway, aren't you testing this against a complete mock cluster provided by `PGTest`? So this should never fail, should it?

Reply:
`PGTest` currently hardcodes the `trust` authentication method, i.e. when using the temporary postgres cluster the test will always fail
https://github.com/jamesnunn/pgtest/blob/417f0d89e66a9395db694a96122e0d5ac0097d59/pgtest/pgtest.py#L506-L507

I will open an issue for this to remind ourselves, also related to https://github.com/aiidateam/aiida-core/issues/3762

Reply:
Correct, see https://github.com/aiidateam/aiida-core/blob/5b6b2b2e2c1d1672e45033dcebf3530626f87bc3/.github/workflows/ci-code.yml#L60

Reply:
We can have a look here at what happens today if we require passwords for postgres
https://github.com/aiidateam/aiida-core/pull/5776